plugins/modules/ndo_physical_interface.py

```diff
@@ -0,0 +1,411 @@
+    type: list
+    elements: dict
+    suboptions:
+      interface_id:
+        description:
+        - The interface ID.
+        type: str
+      description:
+        description:
+        - The description of the interface.
+        type: str
+  state:
+    description:
+    - Use C(absent) for removing.
+    - Use C(query) for listing an object or multiple objects.
+    - Use C(present) for creating or updating.
+    type: str
+    choices: [ absent, query, present ]
+    default: query
+notes:
+- The O(template) must exist before using this module in your playbook.
+  Use M(cisco.mso.ndo_template) to create the Tenant template.
+- The O(physical_policy) must exist before using this module in your playbook.
+  Use M(cisco.mso.ndo_interface_setting) to create the Interface Setting Policy.
+- The O(physical_policy_uuid) must exist before using this module in your playbook.
+  Use M(cisco.mso.ndo_interface_setting) to create the Interface Setting Policy UUID.
+seealso:
+- module: cisco.mso.ndo_template
+extends_documentation_fragment: cisco.mso.modules
+"""
+
+EXAMPLES = r"""
+- name: Create a Physical Interface with physical_interface_type set to physical
+  cisco.mso.ndo_physical_interface:
+  host: mso_host
+  username: admin
+  password: SomeSecretPassword
+  template: ansible_test_template
+  physical_interface: ansible_test_physical_interface_physical
+  description: "Physical Interface for Ansible Test"
+  nodes: [101]
+  interfaces: "1/1"
+  physical_interface_type: physical
+  physical_policy: ansible_test_interface_setting_policy_uuid
+  state: present
+
+- name: Create a Physical Interface with physical_interface_type set to breakout
+  cisco.mso.ndo_physical_interface:
+  host: mso_host
+  username: admin
+  password: SomeSecretPassword
+  template: ansible_test_template
+  physical_interface: ansible_test_physical_interface_breakout
+  description: "breakout interface for Ansible Test"
+  nodes: [101]
+  interfaces: "1/1"
+  physical_interface_type: breakout
+  breakout_mode: 4x25G
+  interface_descriptions:
+    - interface_id: "1/1"
+      description: "Interface description for 1/1"
+  state: present
+
+- name: Query all physical interfaces
+  cisco.mso.ndo_physical_interface:
+  host: mso_host
+  username: admin
+  password: SomeSecretPassword
+  template: ansible_test_template
+  state: query
+  register: query_all
+
+- name: Query a specific Physical Interface with name
+  cisco.mso.ndo_physical_interface:
+  host: mso_host
+  username: admin
+  password: SomeSecretPassword
+  template: ansible_test_template
+  physical_interface: ansible_test_physical_interface_physical
+  state: query
+  register: query_one_name
+
+- name: Query a specific Physical Interface with UUID
+  cisco.mso.ndo_physical_interface:
+  host: mso_host
+  username: admin
+  password: SomeSecretPassword
+  template: ansible_test_template
+  physical_interface_uuid: ansible_test_physical_interface_uuid
```

Comment:
Can we use an actual uuid like below?
"{{ query_one_uuid.current.uuid }}"

Reply:
yes, changed